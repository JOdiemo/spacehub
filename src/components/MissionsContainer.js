import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchData, joinMissions, leaveMission } from '../redux/Missions/MissionsSlice';

const MissionsContainer = () => {
  const data = useSelector((state) => state.data.data);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchData());
  }, [dispatch]);

  return (
    <div className="table-container">
      <table>
        <thead>
          <tr>
            <th>Mission Name</th>
            <th>Mission Description</th>
            <th>Status</th>
            <th>{}</th>
          </tr>
        </thead>
        <tbody>
          {data.map((mission) => (
            <tr key={mission.mission_id}>
              <td>{mission.mission_name}</td>
              <td>{mission.description}</td>
              <td>
                {mission.activeMember && <div className="tag">Active Member</div>}
                {!mission.activeMember && <div className="not-member">Not a Member</div>}
              </td>
              <td>
                <button
                  type="button"
                  onClick={() => (mission.activeMember ? dispatch(leaveMission(mission.mission_id))
                    : dispatch(joinMissions(mission.mission_id)))}
                >
                  {mission.activeMember && <button type="button" className="leave">Leave Mission</button>}
                  {!mission.activeMember && <button type="button" className="join">Join Mission</button>}
                </button>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default MissionsContainer;
