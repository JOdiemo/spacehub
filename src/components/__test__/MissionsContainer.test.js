import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import MissionsContainer from '../MissionsContainer';

// Mock the fetchData function
jest.mock('../../redux/Missions/MissionsSlice', () => ({
  fetchData: jest.fn(),
  joinMissions: jest.fn(),
  leaveMission: jest.fn(),
}));

describe('MissionsContainer', () => {
  const mockStore = configureStore([]);
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      data: {
        data: [
          {
            mission_id: 1, mission_name: 'Mission 1', description: 'Description 1', activeMember: true,
          },
          {
            mission_id: 2, mission_name: 'Mission 2', description: 'Description 2', activeMember: false,
          },
        ],
      },
    });

    component = renderer.create(
      <Provider store={store}>
        <MissionsContainer />
      </Provider>,
    );
  });

  it('should match the snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
