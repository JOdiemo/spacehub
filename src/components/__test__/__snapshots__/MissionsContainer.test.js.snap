// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionsContainer should match the snapshot 1`] = `
<div
  className="table-container"
>
  <table>
    <thead>
      <tr>
        <th>
          Mission Name
        </th>
        <th>
          Mission Description
        </th>
        <th>
          Status
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Mission 1
        </td>
        <td>
          Description 1
        </td>
        <td>
          <div
            className="tag"
          >
            Active Member
          </div>
        </td>
        <td>
          <button
            onClick={[Function]}
            type="button"
          >
            <button
              className="leave"
              type="button"
            >
              Leave Mission
            </button>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Mission 2
        </td>
        <td>
          Description 2
        </td>
        <td>
          <div
            className="not-member"
          >
            Not a Member
          </div>
        </td>
        <td>
          <button
            onClick={[Function]}
            type="button"
          >
            <button
              className="join"
              type="button"
            >
              Join Mission
            </button>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
